source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp

```diff
@@ -0,0 +1,473 @@
+    {
+        const std::vector<std::string> names{ "  Capture time:" };
+        printResults(names, durations);
+    }
+
+    void printSaveResults(const std::vector<std::vector<Duration>> &durations)
+    {
+        const std::vector<std::string> names{ "  Save ZDF:", "  Save PLY:", "  Save PCD:", "  Save XYZ:" };
+        printResults(names, durations);
+    }
+
+    void printZividInfo(Zivid::Application &application)
+    {
+        std::cout << "API: " << Zivid::Version::libraryVersion() << std::endl;
+        auto cameras = application.cameras();
+        if(cameras.size() != 1)
+        {
+            throw std::runtime_error("At least one camera needs to be connected");
+        }
+        std::cout << "Camera: " << cameras.at(0) << std::endl;
+        std::cout << "Compute device: " << cameras.at(0).computeDevice() << std::endl;
+    }
+
+    std::vector<Zivid::Settings> makeSettingsVector(const std::vector<unsigned int> &irises,
+                                                    const std::vector<size_t> &exposureTimes,
+                                                    const bool enableGaussian,
+                                                    const bool enableReflection)
+    {
+        std::vector<Zivid::Settings> settingsVector(irises.size());
+
+        Zivid::Settings::Filters filters;
+        filters.set(Zivid::Settings::Filters::Contrast::Enabled{ true });
+        filters.set(Zivid::Settings::Filters::Outlier::Enabled{ true });
+        filters.set(Zivid::Settings::Filters::Saturated::Enabled{ true });
+        filters.set(Zivid::Settings::Filters::Gaussian::Enabled{ enableGaussian });
+        filters.set(Zivid::Settings::Filters::Reflection::Enabled{ enableReflection });
+
+        Zivid::Settings settings;
+        for(size_t i = 0; i < irises.size(); ++i)
+        {
+            settings.set(Zivid::Settings::Iris{ irises.at(i) });
+            settings.set(Zivid::Settings::Filters{ filters });
+            settings.set(Zivid::Settings::ExposureTime{ std::chrono::microseconds{ exposureTimes.at(i) } });
+            settingsVector.at(i) = settings;
+        }
+        return settingsVector;
+    }
+
+    void benchmarkConnect(Zivid::Application &application, const size_t numConnects)
+    {
+        printConnectHeader(numConnects);
+
+        std::vector<Duration> connectDurations;
+        std::vector<Duration> disconnectDurations;
+        std::vector<std::vector<Duration>> connectDisconnectDurations;
+
+        for(size_t i = 0; i < numConnects; i++)
+        {
+            const auto beforeConnect = HighResClock::now();
+            auto camera = application.connectCamera();
+            const auto afterConnect = HighResClock::now();
+            camera.disconnect();
+            const auto afterDisconnect = HighResClock::now();
+
+            connectDurations.push_back(afterConnect - beforeConnect);
+            disconnectDurations.push_back(afterDisconnect - beforeConnect);
```

Comment:
Was the plan that `disconnectDuration` covers both connection and disconnection? The variable name (to me) would indicate (`afterDisconnect - afterConnect`).  It doesn't matter much, as the information is there. However, calling the variable `totalConnectedDuration` might be more correct? Or, modify so that `disconnectDuration = afterDisconnect - afterConnect`

Reply:
The plan was `disconnectDuration = afterDisconnect `- afterConnect. Fixed now.